legend-engine-external-shared-format-model/src/main/java/org/finos/legend/engine/external/shared/format/model/ExternalFormatModelGenerationExtension.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.external.shared.format.model;
+
+import org.eclipse.collections.api.RichIterable;
+import org.eclipse.collections.api.factory.Lists;
+import org.finos.legend.engine.external.shared.format.model.binding.toModel.SchemaToModelConfiguration;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.shared.core.operational.errorManagement.EngineException;
+import org.finos.legend.pure.generated.CoreGen;
+import org.finos.legend.pure.generated.Root_meta_external_shared_format_ExternalFormatToPureDescriptor;
+import org.finos.legend.pure.generated.Root_meta_external_shared_format_binding_Binding;
+import org.finos.legend.pure.generated.Root_meta_external_shared_format_binding_toPure_SchemaToModelConfiguration;
+import org.finos.legend.pure.generated.Root_meta_external_shared_format_metamodel_SchemaDetail;
+import org.finos.legend.pure.generated.Root_meta_external_shared_format_metamodel_SchemaSet;
+import org.finos.legend.pure.generated.Root_meta_pure_generation_metamodel_GenerationParameter;
+import org.finos.legend.pure.generated.core_pure_generation_generations;
+
+public interface ExternalFormatModelGenerationExtension<Metamodel extends Root_meta_external_shared_format_metamodel_SchemaDetail, ModelGenConfig extends SchemaToModelConfiguration> extends ExternalFormatExtension<Metamodel>
+{
+    /**
+     * Called to compile an external format SchemaToModelConfiguration.
+     */
+    Root_meta_external_shared_format_binding_toPure_SchemaToModelConfiguration compileSchemaToModelConfiguration(ModelGenConfig configuration, PureModel pureModel);
+
+    /**
+     * Provides the properties used to configure model generation.
+     */
+    default RichIterable<? extends Root_meta_pure_generation_metamodel_GenerationParameter> getModelGenerationProperties(PureModel pureModel)
+    {
+        Root_meta_external_shared_format_ExternalFormatToPureDescriptor<?> descriptor = getExternalFormatContract(pureModel)._externalFormatToPureDescriptor();
```

Comment:
Confirm behaviour remains same for flatdata/json/xml/protobuf/grapgQL Introspection